sims/verilator/Makefile

```diff
@@ -1,112 +1,207 @@
+"   compile-debug|debug    = compiles debug simulator [./$(shell basename $(sim_debug))]" \
+"   clean                  = remove all debug/non-debug simulators and intermediate files" \
+"   clean-sim              = removes non-debug simulator and verilator-generated files" \
+"   clean-sim-debug        = removes debug simulator and verilator-generated files"
+
+#############################################################################
+# selectively enabling printfs
+#############################################################################
+ENABLE_PRINTF_PATTERN ?=
+sim_vsrcs_printf := $(sim_vsrcs:.v=.printf.v)
+
+#############################################################################
+# verilator/cxx binary and flags
+#############################################################################
+# can't use -Wall, firrtl spits out 1000's of warnings that we can't fix
 VERILATOR := verilator --cc --exe
 
-CXXFLAGS := \
-	$(CXXFLAGS) -O1 -std=c++11 \
-	-I$(RISCV)/include \
-	-I$(dramsim_dir) \
-	-D__STDC_FORMAT_MACROS \
-	$(EXTRA_SIM_CC_FLAGS)
+#----------------------------------------------------------------------------
+# user configs
+#----------------------------------------------------------------------------
+VERILATOR_PROFILE ?= none
+RUNTIME_PROFILING_CFLAGS := $(if $(filter $(VERILATOR_PROFILE),all),-g -pg,)
+RUNTIME_PROFILING_VFLAGS := $(if $(filter $(VERILATOR_PROFILE),all),\
+                              --prof-threads --prof-cfuncs,\
+                              $(if $(filter $(VERILATOR_PROFILE),threads),\
+                                --prof-threads,))
 
-LDFLAGS := \
-	$(LDFLAGS) \
-	-L$(sim_dir) \
-	-lpthread
+VERILATOR_THREADS ?= 1
+RUNTIME_THREADS := --threads $(VERILATOR_THREADS) --threads-dpi all
 
-VERILATOR_CC_OPTS = \
+VERILATOR_FST_MODE ?= 0
+TRACING_OPTS := $(if $(filter $(VERILATOR_FST_MODE),0),\
+                  --trace,--trace-fst --trace-threads 1)
+TRACING_CFLAGS := $(if $(filter $(VERILATOR_FST_MODE),0),,-DCY_FST_TRACE)
+
+#----------------------------------------------------------------------------
+# verilation configuration/optimization
+#----------------------------------------------------------------------------
+# we initially had --noassert for performance, but several modules use 
+# assertions, including dramsim, so we enable --assert by default
+VMODE := \
 	-O3 \
-	-CFLAGS "$(CXXFLAGS) -DTEST_HARNESS=V$(VLOG_MODEL) -DVERILATOR" \
-	-CFLAGS "-I$(build_dir) -include $(build_dir)/$(long_name).plusArgs -include $(build_dir)/verilator.h" \
-	-LDFLAGS "$(LDFLAGS)" \
-	$(RISCV)/lib/libfesvr.a \
-	$(dramsim_lib)
-
-# default flags added for ariane
-ARIANE_VERILATOR_FLAGS = \
+	--x-assign fast \
+	--x-initial fast \
+	--assert \
+	--output-split 10000 \
+	--output-split-cfuncs 10000
+
+# default flags added for ariane (-Wno-fatal needed for -Wall to not cause
+# a crash, since 1000s of warnings are generated)
+VERILOG_IP_VERILATOR_FLAGS := \
```

Comment:
Why the name change (comment seems to indicate this is still for ariane)?

Reply:
used for nvdla too